File: code/packages/haskell/conduit/src/Conduit.hs
-- Conduit — a Sinatra/Express-style web framework backed by a Rust engine.
--
-- This module re-exports everything a typical Conduit application needs.
-- Import just `Conduit` and you get the full DSL.
--
-- QUICK START
-- -----------
--   import Conduit
--
--   main :: IO ()
--   main = do
--     app <- newApplication
--     get app "/" $ \_ ->
--       return (html 200 "<h1>Hello from Haskell!</h1>")
--     srv <- bind app "127.0.0.1" 8080
--     serve srv

module Conduit
  ( -- * Application
    Application
  , newApplication
  , addRoute
  , get, post, put, delete, patch, options
  , before
  , after
  , notFound
  , onError
  , setSetting
  , getSetting
  , bind

    -- * Server
  , Server
  , serve
  , serveBackground
  , stop
  , localPort
  , running
  , freeServer

    -- * Request
  , Request (..)
  , reqParam
  , reqQuery
  , reqHeader

    -- * Response
  , Response (..)
  , ConduitHalt (..)
  , respond
  , html
  , json
  , textPlain
  , redirect
  , halt
  ) where

import Conduit.App
import Conduit.Server
import Conduit.Request
import Conduit.Response

File: code/packages/dart/conduit/lib/conduit.dart
/// Conduit — Sinatra/Express-style web framework for Dart 3 (WEB17).
///
/// Wraps the conduit-capi Rust cdylib (WEB12) via dart:ffi, providing a
/// fluent object-oriented API:
///
/// ```dart
/// import 'package:coding_adventures_conduit/conduit.dart';
///
/// Future<void> main() async {
///   final server = Application()
///     .get('/', (req) => Response.html('<h1>Hello, Dart!</h1>'))
///     .post('/echo', (req) => Response.text(req.bodyString()))
///     .bind('127.0.0.1', 3000);
///   print('listening on port ${server.localPort}');
///   await server.serve();
/// }
/// ```
library conduit;

export 'src/response.dart' show Response, HaltException;
export 'src/request.dart' show Request;
export 'src/application.dart' show Application;
export 'src/server.dart' show Server;
